test/integration/kubernetes_deploy_test.rb

```diff
@@ -446,6 +446,21 @@ def test_deployment_with_progress_times_out_for_short_duration
     ])
   end
 
+  def test_deployment_with_timeout_override
+    result = deploy_fixtures("long-running", subset: ['undying-deployment.yml.erb']) do |fixtures|
+      deployment = fixtures['undying-deployment.yml.erb']['Deployment'].first
+      deployment['spec']['progressDeadlineSeconds'] = 5
+      deployment["metadata"]["annotations"] = {
+        KubernetesDeploy::KubernetesResource::TIMEOUT_OVERRIDE_ANNOTATION => "10S",
+      }
+      container = deployment['spec']['template']['spec']['containers'].first
+      container['readinessProbe'] = { "exec" => { "command" => ['- ls'] } }
```

Comment:
Why this patch? Doesn't this change mean the pod could actually come up successfully?

Reply:
I don't believe so, since `- ls` will always fail and the container will remain unready. Is there another way to ensure a timeout that I should be using?

Reply:
Oh, I misread this as being an array of commands one of which is `ls`. The bad-probe template uses `["test", "0", "-eq", "1"]` but it doesn't really matter.